### PR TITLE
build: migrate individual DFX workspaces into single root workspace

### PR DESCRIPTION
**Pull request by nathanosdev**
This PR takes all the individual DFX configs for the example projects and merges them into one root config. This makes it much easier to build and run examples and also consolidates the configuration into a single place, making it easier to keep DFX versions and config changes in sync. Additionally makes it easier to make sure that _all_ examples are working at all times.

For each example project:
- Renamed cargo crates for consistency
- Consolidated `dfx.json`, `.gitignore`, `vite.config.ts` and `tsconfig.json` config, `package.json` dependencies to the workspace root
- Updated imports

TT-437

### CODE REVIEW

File: .gitignore
Comment (nathanosdev) [inline .gitignore:1]: All `.gitignore` for example projects are consolidated here now

File: dfx.json
Comment (nathanosdev) [inline dfx.json:1]: All `dfx.json` config is consolidated here now

File: examples/certification/certified-counter/src/backend/Cargo.toml [hunk 1]
```diff
@@ -10,8 +10,7 @@ crate-type = ["cdylib"]
 anyhow.workspace = true
 candid.workspace = true
 ic-cdk.workspace = true
-ic-cdk-macros.workspace = true
-ic-certification.workspace = true
+ic-certification = { workspace = true, features = ["serde"] }
```
Comment (nathanosdev) [inline examples/certification/certified-counter/src/backend/Cargo.toml:13]: This example was previously broken because of this, and the changes in the next file. The changes in this PR mean we will catch this in the CI pipeline next time.

File: examples/certification/certified-counter/src/frontend/.ic-assets.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+[
```
Comment (nathanosdev) [inline examples/certification/certified-counter/src/frontend/.ic-assets.json:1]: New DFX feature means we can use a template for security headers and only specify our headers that will be different from the template. The template's headers match the deleted file, since those headers were taken from ProdSec's recommended list anyway.

File: examples/certification/certified-counter/src/frontend/index.html [hunk 3]
```diff
@@ -5,13 +5,13 @@
     <meta name="viewport" content="width=device-width" />
```
Comment (nathanosdev) [inline examples/certification/certified-counter/src/frontend/index.html:5]: This project was moved from Webpack to Vite to be consistent with the other example projects. Vite is also much simple to configure compared to Webpack. So there are some minor code changes here to work with Vite.

File: examples/response-verification/nodejs/README.md
Comment (nathanosdev) [inline examples/response-verification/nodejs/README.md:1]: All files under `examples/response-verification` were deleted. This examples were just hacky projects thrown together to test the response verification project in various environments during development. Now we have much more extensive e2e and integration tests so these example projects for manual testing haven't actually been used in a long time and they don't serve any purpose to the community either so they're just a maintenance burden at this point.

File: packages/ic-certification/src/hash_tree/mod.rs [hunk 4]
```diff
@@ -34,6 +34,7 @@ impl<Storage: AsRef<[u8]>> Label<Storage> {
     }
 
     /// The length of the output of [`Self::write_hex`]
+    #[cfg(feature = "serde")]
     fn hex_len(&self) -> usize {
```
Comment (nathanosdev) [inline packages/ic-certification/src/hash_tree/mod.rs:38]: Unrelated change. This is unused unless the `serde` feature is enabled and was causing Cargo to throw a warning

File: dfx.json
Comment (nathanosdev) [inline dfx.json:1]: All dfx.json config is consolidated here now

File: examples/response-verification/nodejs/README.md
Comment (nathanosdev) [inline examples/response-verification/nodejs/README.md:1]: All files under examples/response-verification were deleted. This examples were just hacky projects thrown together to test the response verification project in various environments during development. Now we have much more extensive e2e and integration tests so these example projects for manual testing haven't actually been used in a long time and they don't serve any purpose to the community either so they're just a maintenance burden at this point.

File: .github/actions/setup-dfx/action.yml [hunk 5]
```diff
@@ -4,32 +4,8 @@ description: Setup DFX
...
-      with:
-        path: /usr/local/bin/dfx
-        key: dfx-cache-${{ env.dfx_version }}
-
     - name: Install dfx
       uses: dfinity/setup-dfx@main
-      with:
-        dfx-version: '0.18.0'
-
-    - name: Configure DFX
```
Comment (nathanosdev) [inline .github/actions/setup-dfx/action.yml:18]: This is configured in `dfx.json` now

File: .github/actions/setup-dfx/action.yml [hunk 6]
```diff
@@ -4,32 +4,8 @@ description: Setup DFX
...
-    - name: Cache DFX
-      uses: actions/cache@v3
-      with:
-        path: /usr/local/bin/dfx
-        key: dfx-cache-${{ env.dfx_version }}
-
     - name: Install dfx
       uses: dfinity/setup-dfx@main
-      with:
-        dfx-version: '0.18.0'
```
Comment (nathanosdev) [inline .github/actions/setup-dfx/action.yml:16]: Without providing this parameter, the GitHub action will auto-detect the desired `dfx` version from `dfx.json`

File: .github/actions/setup-dfx/action.yml [hunk 7]
```diff
@@ -4,32 +4,8 @@ description: Setup DFX
 runs:
   using: 'composite'
   steps:
-    - name: Cache DFX
-      uses: actions/cache@v3
-      with:
-        path: /usr/local/bin/dfx
-        key: dfx-cache-${{ env.dfx_version }}
```
Comment (nathanosdev) [inline .github/actions/setup-dfx/action.yml:11]: This cache wasn't doing anything. `dfxvm` will still download and install new `dfx`. The caching is leftover from the approach we were using before the `dfinity/setup-dfx` action was released.

File: examples/certification/certified-counter/src/backend/src/lib.rs [hunk 8]
```diff
@@ -1,16 +1,15 @@
-use ic_cdk::export::candid::CandidType;
+use candid::CandidType;
```
Comment (nathanosdev) [inline examples/certification/certified-counter/src/backend/src/lib.rs:1]: This example project was broken because of this change, which is part of the motivation for these changes

File: examples/certification/certified-counter/src/frontend/.ic-assets.json
Comment (nathanosdev) [inline examples/certification/certified-counter/src/frontend/.ic-assets.json:1]: The `certified-counter` example has more changes than the others because I migrated away from Webpack to Vite as the build system for the frontend. This makes it consistent with all of the other example projects.

File: examples/response-verification/nodejs/README.md
Comment (nathanosdev) [inline examples/response-verification/nodejs/README.md:1]: I deleted the `response-verification` examples because they were hacky and don't provide anything useful for community members. They used to serve as a manual testing ground, but since there's many e2e automation tests now, they haven't actually been used for manual testing purposes for a long time, so they're just a maintenance burden now.

File: packages/ic-response-verification-tests/README.md
Comment (nathanosdev) [inline packages/ic-response-verification-tests/README.md:1]: This e2e test project was also migrated to use Vite instead of Webpack for consistency, and I also flattened the structure a bit. It used to be JS tests and a canister nested inside a Cargo project, now it's 3 siblings inside a normal directory.

**Review by przydatek — APPROVED** [review]
LGTM, thanks!
